controllers/rack.go

```diff
@@ -613,11 +615,16 @@ func (r *SingleClusterReconciler) scaleDownRack(
 
 	// Ignore safe stop check on pod not in running state.
 	if utils.IsPodRunningAndReady(pod) {
-		if res := r.waitForMultipleNodesSafeStopReady([]*corev1.Pod{
-			pod}, ignorablePods); !res.isSuccess {
+		if res := r.waitForMultipleNodesSafeStopReady([]*corev1.Pod{pod}, ignorablePods); !res.isSuccess {
 			// The pod is running and is unsafe to terminate.
 			return found, res
 		}
+
+		// Setup roster after migration.
+		if err := r.getAndSetRoster(policy, r.aeroCluster.Spec.RosterBlockList); err != nil {
+			r.Log.Error(err, "Failed to set roster for cluster")
+			return found, reconcileRequeueAfter(0)
+		}
```

Comment:
Didn't understand this. Why are we setting the roster here instead of setting it after `validateSCClusterState `  below?

Reply:
We are waiting for the migration before setting the roster. This set roster call is for the previous iteration of scale-down. The last scale-down iteration is handled in the main reconcile function.